src/test/extensions/updateHelper.spec.ts

```diff
@@ -31,177 +33,465 @@ const SPEC = {
+      isOfficialStub = sinon.stub(resolveSource, "isOfficialSource");
+      isOfficialStub.returns(false);
+      getInstanceStub = sinon.stub(extensionsApi, "getInstance").resolves(REGISTRY_INSTANCE);
     });
 
-    it("should prompt for changes to roles and continue if user gives consent", () => {
+    afterEach(() => {
+      promptStub.restore();
+      getExtensionStub.restore();
+      createSourceStub.restore();
+      registryStub.restore();
+      isOfficialStub.restore();
+      getInstanceStub.restore();
+    });
+    it("should return the correct source name for a valid published extension version source", async () => {
       promptStub.resolves(true);
-      const newSpec = _.cloneDeep(SPEC);
-      newSpec.roles = [
-        { role: "role2", reason: "" },
-        { role: "role3", reason: "" },
-      ];
-
-      expect(updateHelper.displayChangesRequiringConfirmation(SPEC, newSpec)).not.to.be.rejected;
-
-      expect(promptStub.callCount).to.equal(1);
-      expect(promptStub.firstCall.args[0].message).to.contain("- role1");
-      expect(promptStub.firstCall.args[0].message).to.contain("+ role3");
+      getExtensionStub.resolves(EXTENSION);
+      createSourceStub.resolves(EXTENSION_VERSION);
+      const name = await updateHelper.updateToVersionFromPublisherSource(
+        "test-project",
+        "test-instance",
+        "test-publisher/test@0.2.0",
+        SPEC,
+        SPEC.name
+      );
+      expect(name).to.equal(EXTENSION_VERSION.name);
     });
-
-    it("should prompt for changes to resources and continue if user gives consent", () => {
+    it("should throw an error for an invalid source", async () => {
       promptStub.resolves(true);
-      const newSpec = _.cloneDeep(SPEC);
-      newSpec.resources = [
-        { name: "resource3", type: "firebaseextensions.v1beta.function", description: "new desc" },
-        { name: "resource2", type: "other", description: "" },
-      ];
-
-      expect(updateHelper.displayChangesRequiringConfirmation(SPEC, newSpec)).not.to.be.rejected;
+      getExtensionStub.throws(Error("NOT FOUND"));
+      createSourceStub.throws(Error("NOT FOUND"));
+      await expect(
+        updateHelper.updateToVersionFromPublisherSource(
+          "test-project",
+          "test-instance",
+          "test-publisher/test@1.2.3",
+          SPEC,
+          SPEC.name
+        )
+      ).to.be.rejectedWith("NOT FOUND");
+    });
+    it("should not update if the update warning is not confirmed", async () => {
+      promptStub.resolves(false);
+      getExtensionStub.resolves(EXTENSION);
+      createSourceStub.resolves(EXTENSION_VERSION);
+      await expect(
+        updateHelper.updateToVersionFromPublisherSource(
+          "test-project",
+          "test-instance",
+          "test-publisher/test@0.2.0",
+          SPEC,
+          SPEC.name
+        )
+      ).to.be.rejectedWith(FirebaseError, "Update cancelled.");
+    });
+  }).timeout(2500);
```

Comment:
Another suspicious timeout

Reply:
Removed it and tests still pass, thanksfully!